source/iopipe/json/dom.d

```diff
@@ -7,91 +7,211 @@ import iopipe.json.parser;
 public import iopipe.json.common;
 import iopipe.traits;
 import std.traits;
+import std.conv : to;
 
 enum JSONType
 {
-    Integer,
-    Floating,
+    Integer,        // Now a hint only, stored as string
+    Floating,       // Now a hint only, stored as string
     String,
     Obj,
     Array,
     Null,
     Bool,
+    StringSSO,      // Small string optimization for strings
+    NumberSSO,      // Small string optimization for numbers
 }
 
-struct JSONValue(SType)
+// Making JSONValue a template struct with a default parameter of string
+struct JSONValue(StringType = string)
 {
-    // basically a tagged union.
+    // Tagged union
     JSONType type;
+    
+    // Small string optimization - 16 bytes should cover many common cases
+    private enum SSOSize = 16;
+    
     union
     {
-        long integer;
-        real floating;
+        // String storage (either SSO or allocated)
+        struct {
+            union {
+                char[SSOSize] sso;
+                StringType allocatedString;
+            }
+            // For SSO types, first byte can indicate length used
+            ubyte ssoLength;
```

Comment:
This is OK, for now. My suggestion was to use the type field to tell whether the length was used or not. I can fix this up later.

The goal is to maximize SSO size without adding to the size of the struct. In reality, I think we can pack this whole struct tighter.